hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/container/TestHelper.java

```diff
@@ -234,12 +235,14 @@ public static void waitForPipelineClose(List<Pipeline> pipelineList,
 
     // wait for the pipeline to get destroyed in the datanodes
     for (Pipeline pipeline : pipelineList) {
+      HddsProtos.PipelineID pipelineId = pipeline.getId().getProtobuf();
       for (DatanodeDetails dn : pipeline.getNodes()) {
         XceiverServerSpi server =
             cluster.getHddsDatanodes().get(cluster.getHddsDatanodeIndex(dn))
                 .getDatanodeStateMachine().getContainer().getWriteChannel();
         Assert.assertTrue(server instanceof XceiverServerRatis);
-        server.removeGroup(pipeline.getId().getProtobuf());
+        GenericTestUtils.waitFor(() -> !server.isExist(pipelineId),
+            100, 30_000);
```

Comment:
Thanx @adoroszlai for the fix, I am able to repro this, changes seems fair enough.
Does it makes sense to `removeGroup` explicitly in case `server.isExist(pipelineId)` is true, rather than waiting? Might speed up test? We can ignore the exception from `removeGroup` in the wait and retry to counter the race condition

Reply:
Thanks @ayushtkn for taking a look.

I'd rather avoid such "manual intervention" and let the test exercise production code paths.

I don't think keeping `removeGroup` would result in much speedup.  If SCM and datanodes take too much time to execute regular flow, we could probably tweak executor intervals, etc.

It might be possible to convert it to a unit test with mocks, resulting in much bigger speedup by avoiding the mini cluster.

Reply:
makes better to stay near the prod code. Thanx